client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -160,31 +160,20 @@ class CalypsoifyIframe extends Component<
 	mediaCancelPort: MessagePort | null = null;
 	revisionsPort: MessagePort | null = null;
 	checkoutPort: MessagePort | null = null;
-	successfulIframeLoad = false;
-	waitForIframeToInit: ReturnType< typeof setInterval > | undefined = undefined;
-	waitForIframeToLoad: ReturnType< typeof setTimeout > | undefined = undefined;
 
 	componentDidMount() {
 		window.addEventListener( 'message', this.onMessage, false );
 
-		const isDesktop = config.isEnabled( 'desktop' );
-		// If the iframe fails to load for some reson, eg. an unexpected auth loop, this timeout
-		// provides a redirect to wpadmin for web users - this should now be a rare occurance with
-		// a 3rd party cookie auth issue fix in place https://github.com/Automattic/jetpack/pull/16167
-		this.waitForIframeToInit = setInterval( () => {
+		// If the iframe itself fails to load for some reason, eg. an unexpected
+		// auth loop, we need to exit to wp-admin without the iFrame. Therefore,
+		// once we start loading the iFrame, we set the editor redirect timer to
+		// 25s. This timer is cleared when iFrame.onLoad is called.
+		const waitForIframeToInit = setInterval( () => {
 			if ( this.props.shouldLoadIframe ) {
```

Comment:
If `this.props.shouldLoadIframe === false` we potentially never clear the interval?

Maybe we should consider moving the `if ( this.props.shouldLoadIframe ) {` check outside the `setInterval` call?

Reply:
I'm not sure what the original intent for this code was, but I think the existing logic &: to basically run the interval _until_ `shouldLoadIframe` is true. In other words, we don't want to start a countdown if shouldLoadIframe is false, but we do as soon as it's true.

If this were a functional component, it'd be a good candidate for useEffect. I'm not sure if componentDidUpdate would replicate it entirely

Reply:
Just FYI, the [`sandbox`](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/components/src/sandbox/index.js) component in the core might be a good inspiration for component structure.

Reply:
I've refactored this into didMount/didUpdate. Hopefully we can refactor this to a FC in the future